test/integration/targets/docker_swarm_service/tasks/tests/configs.yml

```diff
@@ -64,20 +63,33 @@
       - config_id: "{{ config_result_1.config_id|default('') }}"
         config_name: "{{ config_name_1 }}"
         filename: "/tmp/{{ config_name_1 }}.txt"
-      - config_id: "{{ config_result_2.config_id|default('') }}"
-        config_name: "{{ config_name_2 }}"
+      - config_name: "{{ config_name_2 }}"
         filename: "/tmp/{{ config_name_2 }}.txt"
   register: configs_3
   ignore_errors: yes
 
+- name: configs (add idempotency)
+  docker_swarm_service:
+    name: "{{ service_name }}"
+    image: alpine:3.8
+    resolve_image: no
+    command: '/bin/sh -v -c "sleep 10m"'
+    configs:
+      - config_name: "{{ config_name_1 }}"
+        filename: "/tmp/{{ config_name_1 }}.txt"
+      - config_name: "{{ config_name_2 }}"
```

Comment:
Maybe add the ID here? Then in one of the add/idempotency pairs, you once have the ID in the first but not in the second, and once you have it in the second but not in the first.

Reply:
That's a good idea. I wanted the test to run with two non id configs / secrets so we test passing multiple items as filters to the Docker API. I added a new test task for this.